src/canvas/heatmapFeature.js

```diff
@@ -124,54 +124,23 @@ var canvas_heatmapFeature = function (arg) {
    * @protected
    */
   ////////////////////////////////////////////////////////////////////////////
-  this._colorize = function (context2d, width, height, imageData, gradient) {
-    var isLittleEndian = true, i, j, index, pixels;
-
-    // Determine whether Uint32 is little- or big-endian.
-    if (!m_typedBuffer || (m_typedBuffer.length !== imageData.data.length)) {
-      m_typedBuffer = new ArrayBuffer(imageData.data.length);
+  this._colorize = function (pixels, gradient) {
+    var grad = new Uint32Array(gradient.buffer);
+    var i, j, k, pixlen = pixels.length;
```

Comment:
The one `var` rule comes from jslint.  I think the justification for it is preventing C programmers from misunderstanding how scoping rules work in javascript.  My preference is just to keep all of the variable declarations at the top of the function, but I don't think enforcing a one `var` rule is worthwhile.